src/Symfony/Component/Yaml/Inline.php

```diff
@@ -269,7 +269,7 @@ private static function dumpNull(int $flags): string
      */
     public static function parseScalar(string $scalar, int $flags = 0, array $delimiters = null, int &$i = 0, bool $evaluate = true, array $references = [])
     {
-        if (\in_array($scalar[$i], ['"', "'"])) {
+        if ('"' === $scalar[$i] || "'" === $scalar[$i]) {
```

Comment:
seems not to be faster: https://3v4l.org/ltQqN

Reply:
You're right, this was a pitfall: the string-comparison method is comparable / beats in_array (and doesn't add a function call to trace) if the first character is a match - but this probably isn't the most frequent use case. I'll undo this.

Reply:
switching to a strict `in_array` might help though

Reply:
@stof I did exactly that for the cases where it is possible, e.g. all places checking if char is a quote.